services/placement/algo/helper.go

```diff
@@ -385,16 +418,79 @@ func (h *instanceHeap) Pop() interface{} {
 	return instance
 }
 
-func addInstanceToPlacement(p services.ServicePlacement, i services.PlacementInstance) (services.ServicePlacement, services.PlacementInstance, error) {
+// MarkShardAvailable marks the state of a shard to available
+func MarkShardAvailable(p services.ServicePlacement, instanceID string, shardID uint32) (services.ServicePlacement, error) {
```

Comment:
Should this method perhaps live on the `PlacementService`?  It's not extremely discoverable as a method sitting in the `algo` package no?